demos/link-prediction/homogeneous-comparison-link-prediction.ipynb

```diff
@@ -0,0 +1,1634 @@
+{
```

Comment:
<p>You need to use a single cell with all of the parameters, as the new parameters are inserted after the first one, and thus the later ones override it. You can see this in the notebook that was run on CI: https://nbviewer.jupyter.org/urls/buildkite.com/organizations/stellar/pipelines/stellargraph-public/builds/4791/jobs/292b124f-0618-4cab-9b0a-93762e76243e/artifacts/929ec060-09dd-4294-abf7-c8b79fd6c07b#Suggested-Hyperparameters</p><p> </p><p>The inserted cell is the one with the <code># Parameters</code>  comment, and the three later cells win and override those customisations.</p><p> </p><p>(I think the failure on CI is due to running out of memory; hopefully that will be addressed by getting these parameters to be overridden correctly!)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stellargraph/stellargraph/pull/1658/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='27'/>

Reply:
Thanks @huonw ! I ever try to use a single cell to make the variables overwritten by the notebook-parameters.yml. However, one the batch_size is overwritten to 5, it will consume more time. In the new update, I only make the `epochs` variable be overwritten. I set different `walk_number` values for different algorithms. With the overwritten the `epochs`, the demo will consume much less time.

Reply:
Thanks @huonw , I will change the cell carefully to make the parameter correctly overwritten by notebook-parameters.yml.

Reply:
To ensure that `batch_size = 50` "wins", that cell will need to be after the `epoch` parameters one, see the ordering in: https://nbviewer.jupyter.org/urls/buildkite.com/organizations/stellar/pipelines/stellargraph-public/builds/4798/jobs/3253fa74-8196-4f0e-a480-5e22a206f48d/artifacts/54d6f390-7c39-4d63-9033-ca62d3466431#Parameter-Settings

Reply:
Yeah, I have changed the order.